### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by katherinemtam**

### CODE REVIEW

File: craco.config.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+module.exports = {
+  style: {
+    postcss: {
+      plugins: [
+        require('tailwindcss'),
+        require('autoprefixer'),
+      ],
+    },
+  },
+}
```
Comment (dbennin125) [inline craco.config.js:11]: new line

File: src/components/search/SearchPage.jsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      return prevPage - 1
+    })
+    const artists = await fetchArtists(search, page);
+    setArtists(artists);
+  };
+
+  return (
+    <>
+      <SearchForm
+        onInputChange={handleInputChange}
```
Comment (dbennin125) [inline src/components/search/SearchPage.jsx:42]: you can do this is inline if you'd like. 
onInputChange={({target} => setSearch(target.value))}

File: src/components/search/SearchPage.jsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    e.preventDefault();
+    const artists = await fetchArtists(search, page);
+    setArtists(artists);
+  }
+
+  const handleNextClick = async () => {
+    setPage((prevPage) => {
+      return prevPage + 1
+    })
+    const artists = await fetchArtists(search, page);
```
Comment (dbennin125) [inline src/components/search/SearchPage.jsx:27]: lines 27-29 (and also 35 and 36) should be a in  useEffect. Making a fetch call? think useEffect. Need it to change when a page changes? useEffect with a [pageState].

File: src/services/searchApi.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
+import parser from 'fast-xml-parser';
+
+export const fetchArtists = async (search, page) => {
+  const res = await fetch(`http://musicbrainz.org/ws/2/artist?query=${search}&fmt=json&limit=100`);
```
Comment (dbennin125) [inline src/services/searchApi.js:4]: you'll want to change this to https

File: src/services/searchApi.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  const json = await res.json();
+  return json.artists.slice((page - 1) * 10, page * 10);
+}
+
+export const fetchReleases = async (id, page) => {
+  const res = await fetch(`http://musicbrainz.org/ws/2/release?artist=${id}&fmt=json`);
```
Comment (dbennin125) [inline src/services/searchApi.js:10]: https

File: src/services/searchApi.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  const json = await res.json();
+  return json.releases.slice((page - 1) * 5, page * 5);
+}
+
+export const fetchRecordings = async (id) => {
+  const res = await fetch(`http://musicbrainz.org/ws/2/recording?release=${id}&fmt=json`)
```
Comment (dbennin125) [inline src/services/searchApi.js:16]: https

File: src/services/searchApi.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  const json = await res.json();
+
+  return json.recordings;
+}
+
+export const fetchSong = async (artist, title) => {
```
Comment (dbennin125) [inline src/services/searchApi.js:22]: This function didn't work for me, but super cool that you went the extra mile.

File: src/components/releases/ReleasesList.jsx [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  const { artistName, artistId } = useParams();
+
+  const [releases, setReleases] = useState([])
+  const [page, setPage] = useState(1)
+
+  const handleNextClick = async () => {
+    setPage((prevPage) => {
+      return prevPage + 1
+    })
+    const releases = await fetchReleases(artistId, page);
```
Comment (dbennin125) [inline src/components/releases/ReleasesList.jsx:17]: fetch call? think useEffect.

**Review by dbennin125 — COMMENTED** [review]
Great work! Couldn't get the lyrics to display but great work regardless! Awesome job!

**Comment by katherinemtam** [comment]
> Great work! Couldn't get the lyrics to display but great work regardless! Awesome job!

Thanks for your comments! We ended up changing http to https in later PRs. One of the APIs did not work with https though.  Also, the lyrics API is rather outdated, so lyrics may or may not appear based on the API.

**Pull request merged**